server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/VariableWidthHistogramAggregatorFactory.java

```diff
@@ -0,0 +1,94 @@
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+import org.elasticsearch.search.aggregations.support.AggregatorSupplier;
+import org.elasticsearch.search.aggregations.support.CoreValuesSourceType;
+import org.elasticsearch.search.aggregations.support.ValuesSource;
+import org.elasticsearch.search.aggregations.support.ValuesSourceAggregatorFactory;
+import org.elasticsearch.search.aggregations.support.ValuesSourceConfig;
+import org.elasticsearch.search.aggregations.support.ValuesSourceRegistry;
+import org.elasticsearch.search.internal.SearchContext;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class VariableWidthHistogramAggregatorFactory extends ValuesSourceAggregatorFactory {
+
+    public static void registerAggregators(ValuesSourceRegistry.Builder builder) {
+        builder.register(VariableWidthHistogramAggregationBuilder.NAME, CoreValuesSourceType.NUMERIC,
+            (VariableWidthHistogramAggregatorSupplier) VariableWidthHistogramAggregator::new);
+    }
+
+    private final int numBuckets;
+    private final int shardSize;
+    private final int initialBuffer;
+
+    VariableWidthHistogramAggregatorFactory(String name,
+                                            ValuesSourceConfig config,
+                                            int numBuckets,
+                                            int shardSize,
+                                            int initialBuffer,
+                                            QueryShardContext queryShardContext,
+                                            AggregatorFactory parent,
+                                            AggregatorFactories.Builder subFactoriesBuilder,
+                                            Map<String, Object> metadata) throws IOException{
+        super(name, config, queryShardContext, parent, subFactoriesBuilder, metadata);
+        this.numBuckets = numBuckets;
+        this.shardSize = shardSize;
+        this.initialBuffer = initialBuffer;
+    }
+
+    @Override
+    protected Aggregator createUnmapped(SearchContext searchContext,
+                                        Aggregator parent,
+                                        Map<String, Object> metadata) throws IOException {
+        return new VariableWidthHistogramAggregator(name, factories, numBuckets, shardSize, initialBuffer, null,
+            config.format(), searchContext, parent, metadata);
+    }
+
+
+    @Override
+    protected Aggregator doCreateInternal(ValuesSource valuesSource,
+                                          SearchContext searchContext,
+                                          Aggregator parent,
+                                          boolean collectsFromSingleBucket,
+                                          Map<String, Object> metadata) throws IOException {
+        if (collectsFromSingleBucket == false) {
+            return asMultiBucketAggregator(this, searchContext, parent);
```

Comment:
You are just about the last user of this method. Once this PR is merged, I'd like to open a follow up PR to remove this. Would you be willing to review it when I do? I mean, if you want to do it you can, but I don't want to block this PR. Its been long enough :)

Reply:
No worries, I took care of it. It was a really easy fix after I merged master in.

Before I push my changes I just wanted to confirm something: I merged without squashing first. Will that be okay?